packages/amplify-graphql-transformer-core/src/transformer-context/resolver.ts

```diff
@@ -0,0 +1,280 @@
+    const requestFns = this.synthesizePipelineFunctions(stack, api, this.requestSlots);
+    const responseFns = this.synthesizePipelineFunctions(stack, api, this.responseSlots);
+    // substitue template name values
+    [this.requestMappingTemplate, this.requestMappingTemplate].map(template => this.substitueSlotInfo(template, 'main', 0));
+
+    const dataSourceProviderFn = api.addAppSyncFunction(
+      toPascalCase([this.typeName, this.fieldName, 'DataResolverFn']),
+      this.requestMappingTemplate,
+      this.responseMappingTemplate,
+      this.datasource?.name || NONE_DATA_SOURCE_NAME,
+      stack,
+    );
+
+    let dataSourceType = 'NONE';
+    let dataSource = '';
+    if (this.datasource) {
+      dataSourceType = this.datasource.ds.type;
+      switch (dataSourceType) {
+        case 'AMAZON_DYNAMODB':
+          if (this.datasource.ds.dynamoDbConfig && !isResolvableObject(this.datasource.ds.dynamoDbConfig)) {
+            const tableName = this.datasource.ds.dynamoDbConfig?.tableName;
+            dataSource = `$util.qr($ctx.stash.put("tableName", "${tableName}"))`;
+          }
+          break;
+        case 'AMAZON_ELASTICSEARCH':
+          if (this.datasource.ds.elasticsearchConfig && !isResolvableObject(this.datasource.ds.elasticsearchConfig)) {
+            const endpoint = this.datasource.ds.elasticsearchConfig?.endpoint;
+            dataSource = `$util.qr($ctx.stash.put("endpoint", "${endpoint}"))`;
+          }
+          break;
+        case 'AWS_LAMBDA':
+          if (this.datasource.ds.lambdaConfig && !isResolvableObject(this.datasource.ds.lambdaConfig)) {
+            const lambdaFunctionArn = this.datasource.ds.lambdaConfig?.lambdaFunctionArn;
+            dataSource = `$util.qr($ctx.stash.put("lambdaFunctionArn", "${lambdaFunctionArn}"))`;
+          }
+          break;
+        case 'HTTP':
+          if (this.datasource.ds.httpConfig && !isResolvableObject(this.datasource.ds.httpConfig)) {
+            const endpoint = this.datasource.ds.httpConfig?.endpoint;
+            dataSource = `$util.qr($ctx.stash.put("endpoint", "${endpoint}"))`;
+          }
+          break;
+        case 'RELATIONAL_DATABASE':
+          if (
+            this.datasource.ds.relationalDatabaseConfig &&
+            !isResolvableObject(this.datasource.ds.relationalDatabaseConfig) &&
+            !isResolvableObject(this.datasource.ds.relationalDatabaseConfig?.rdsHttpEndpointConfig)
+          ) {
+            const databaseName = this.datasource.ds.relationalDatabaseConfig?.rdsHttpEndpointConfig!.databaseName;
+            dataSource = `$util.qr($ctx.stash.metadata.put("databaseName", "${databaseName}"))`;
+          }
+          break;
+        default:
+          throw new Error('Unknow DataSource type');
+      }
+    }
+    api.addResolver(
+      this.typeName,
+      this.fieldName,
+      MappingTemplate.inlineTemplateFromString(
+        dedent`
+      $util.qr($ctx.stash.put("typeName", "${this.typeName}"))
```

Comment:
seems out of place to have VTL defined here. I would at least put the VTL generation in a separate helper function